input/dir/dir_input_test.go

```diff
@@ -0,0 +1,103 @@
+package dir
+
+import (
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"testing"
+	"time"
+
+	. "github.com/smartystreets/goconvey/convey"
+	"github.com/spacemonkeygo/errors"
+	"github.com/spacemonkeygo/errors/try"
+	"polydawn.net/repeatr/def"
+	"polydawn.net/repeatr/lib/fshash"
+	"polydawn.net/repeatr/testutil"
+)
+
+func Test(t *testing.T) {
+	Convey("Given a nonexistant path", t, func() {
+		Convey("The input config should be rejected during validation", func() {
+			correctError := false
+			try.Do(func() {
+				New(def.Input{
+					Type:     "dir",
+					Hash:     "abcd",
+					URL:      "/tmp/certainly/should/not/exist",
+					Location: "/data",
+					// REVIEW: two of these four fields were more for the executor.
+					// maybe this struct isn't the sanest to use for this arg.
+				})
+			}).Catch(def.ValidationError, func(e *errors.Error) {
+				correctError = true
+			}).Done()
+
+			So(correctError, ShouldBeTrue)
+		})
+	})
+
+	Convey("Given a directory with a mixture of files and folders", t,
+		testutil.WithTmpdir(func() {
+			pwd, _ := os.Getwd()
+			os.Mkdir("src", 0755)
+			os.Mkdir("src/a", 01777)
```

Comment:
Yup.  I just picked an amusing set of numbers; all of which are covered by the assertions that load a tar header, since that includes mode bits.